plugins/CoreHome/angularjs/siteselector/siteselector-model.service.js

```diff
@@ -63,17 +64,31 @@
             }
 
             if (model.isLoading) {
-                model.currentRequest.abort();
+                if (model.currentRequest) {
+                    model.currentRequest.abort();
+                } else if (limitPromise) {
+                    limitPromise.abort();
+                    limitPromise = null;
+                }
             }
 
             model.isLoading = true;
 
-            model.currentRequest = piwikApi.fetch({
-                method: 'SitesManager.getPatternMatchSites',
-                pattern: term
-            });
+            if (!limitPromise) {
+                limitPromise = piwikApi.fetch({method: 'SitesManager.getNumWebsitesToDisplayPerPage'});
+            }
```

Comment:
Instead of doing this request here, would it be better to allow specifying the limit as a HTML attribute in the directive? Then it could default to the result of this API call. I think since the result of this call is used to change how much data the site selector displays and not what it displays, it should be handled by the directive or controller.